proposals/0061-fair-congestion-control.md

```diff
@@ -0,0 +1,258 @@
+
+To that end, it introduces a exponentially-scaled dynamic base fees to
+individual local fee markets. It also attains very short feedback loop of
+per-tx frequency to maintain full efficacy of Solana's peculiar execution
+model compared to other blockchains: multi-threaded and low latency.
+
+This is realized by means of some incentive tweaks to combat against the
+obvious base fee manipulation with such short intervals.
+
+## Motivation
+
+(still in half-finished...)
+
+- Write lock cu limit is bad (bot can lock out at the very first of block for
+  the entire duration of whole blocktime (400ms)
+- Increased Defi activities around any volatile financial markets could starve
+  payment transactions for extended time
+- Inter-block and linear Voluntary fee escalation with vanilla fee market
+  auction can't guarantee the scheduling deadline of casual payment txes (which
+  needs 99.99% sub-second confirmation at very minimum).
+
+## Overview of the Design
+
+This proposal tries to localize congestions by means of increasing minimum
+required `cu_price`s for each write-locked addresses (= `base_cu_price`), which
+is newly introduced by this proposal. This increase will be calculated
+exponentially, measured by the CU consumed by each addresses at the moment.
+This means a transaction must cost the sum of `requested_cu * base_cu_price`
+for all of its write-locked addresses at least. This results in selectively
+pricing out crowded subset of transactions waiting for block inclusion, while
+allowing other transactions to be processed for block inclusion.
+
+In this way, under-capitalized usage demand like spl-token transfers among
+wallets will still find its way for _timely_ block inclusion, accomplishing
+this mechanism's advertised _fairness_ in the sense of blockspace allocation
+and transaction fee isolation for each independent local fee markets even at
+the time of financial market's volatility, which entices well-capitalized usage
+demand in DeFi activities on chain. 
+
+This rate-limiting gets enforced only when the cluster deemed to be congested,
+meaning no idling block space is wasted when not congested. Also, those
+exponentially-increased `base_cu_price`s will be decreased exponentially
+likewise in terms of consumed CUs in a block (= `block_cu`), as long as its
+address-specific CUs (= `address_cu`) remain to be unchanged due to the
+temporal stalemate of relevant transaction processing. Collectively, each
+addresses get equal amount of opportunity to execute transactions in
+round-robin fashion while contributing to the decrease of other
+`base_cu_price`s, assuming no exponential priority-fee premium is paid among
+users to interrupt the orderly processing with such a high economical
+justification.
+
+On top of the direct appreciation of aforementioned fairness, this proposal
+also obsoletes both the existing block-wide CU limit and the account-write CU
+limit to overcome their inherent unfairness and problems. Also, no global
```

Comment:
Will the dynamic `base-cu-price` fully safe-guard reply timeliness (eg <=400ms)? Is it still somewhat desirable to have a hard limit to be safe?